### DISCUSSION TITLE
Supporting S3 driver for uploading results

### DISCUSSION THREAD

**Discussion by jojomak13** [gotenberg/gotenberg#1084] [Q&A] [answered]
I was wondering if we could support uploading the PDF result directly to S3, especially in the case of using the webhook feature.

I would like to help but need your guidance on how to get started with the codebase.

**Comment by gulien** [comment, marked as answer]
Hey @jojomak13,

You can already use the webhook feature with S3, a lot of user have reported using it 😄

I don’t have an example to give you, but I guess you can find a way with the webhook headers.

**Reply by jojomak13** [reply]
If you could provide me with example, would be helpful a lot

**Reply by Morzaram** [reply]
The way I did it was generate a presign URL, and added the following headers
```
  { 
    'Gotenberg-Webhook-Url': presign_url,
    'Gotenberg-Webhook-Method': 'PUT'
    'Gotenberg-Webhook-Error-Url': 'https://my-error-handling-url.com/handle_error'
  }
```
